src/DataStreams/ParallelParsingBlockInputStream.cpp

```diff
@@ -195,18 +199,24 @@ void ParallelParsingBlockInputStream::parserThreadFunction(ThreadGroupStatusPtr
     }
     catch (...)
     {
-        onBackgroundException();
+        onBackgroundException(unit.offset);
     }
 }
 
-void ParallelParsingBlockInputStream::onBackgroundException()
+void ParallelParsingBlockInputStream::onBackgroundException(size_t offset)
 {
     tryLogCurrentException(__PRETTY_FUNCTION__);
 
     std::unique_lock<std::mutex> lock(mutex);
     if (!background_exception)
     {
         background_exception = std::current_exception();
+ 
+        if (Exception * e = exception_cast<Exception *>(background_exception))
+            e->addMessage(fmt::format(
+                "Offset: {}. P.S. You have to sum up offset and the row number from"
```

Comment:
'Offset' is an ambiguous word -- bytes or rows? Let's use 'row number'. The message is somewhat unfriendly -- like we've given up on calculating the offset in our code and now require the user to do this manually. Maybe we can do this automatically? One way is to add the row number as separate field for parsing exceptions. Another is to pass the 'true' row number inside the parser. Not sure what's more convenient.

Reply:
> Another is to pass the 'real' row number inside the parser.
What does it mean? Each instance of parser knows nothing about parallel execution.

I think the first variant it Ok.